include/nuttx/spinlock.h

```diff
@@ -39,6 +39,26 @@
 typedef uint8_t spinlock_t;
 #else
 
+#ifdef CONFIG_TICKET_SPINLOCK
+
+#  define SP_UNLOCKED 0  /* The Un-locked state */
+#  define SP_LOCKED   1  /* The Locked state */
+
+#  define INIT_SPINLOCK_VALUE {{SP_UNLOCKED}}
+
+union spinlock_u
+{
+  struct
+  {
+    uint16_t owner;
+    uint16_t next;
+  } tickets;
```

Comment:
ticket

Reply:
follow the (path)[https://lore.kernel.org/linux-riscv/CAAhSdy1JHLUFwu7RuCaQ+RUWRBks2KsDva7EpRt8--4ZfofSUQ@mail.gmail.com/T/].

Using tickets means owner and next.

Reply:
ok